geth/api/backend.go

```diff
@@ -77,120 +80,145 @@ func (m *StatusBackend) IsNodeRunning() bool {
+					Error: fmt.Errorf("%v: %v", node.ErrNodeStartFailure, err).Error(),
+				},
+			})
+		default:
+			signal.Send(signal.Envelope{
+				Type: signal.EventNodeCrashed,
+				Event: signal.NodeCrashEvent{
+					Error: fmt.Errorf("Node creation failed: %v", err).Error(),
+				},
+			})
+		}
+
+		return err
 	}
-
-	m.nodeReady = make(chan struct{}, 1)
-	go m.onNodeStart(nodeStarted, m.nodeReady) // waits on nodeStarted, writes to backendReady
-
-	return m.nodeReady, err
+	signal.Send(signal.Envelope{
+		Type:  signal.EventNodeStarted,
+		Event: struct{}{},
+	})
+	m.onNodeStart()
+	return nil
 }
 
 // onNodeStart does everything required to prepare backend
-func (m *StatusBackend) onNodeStart(nodeStarted <-chan struct{}, backendReady chan struct{}) {
-	<-nodeStarted
-
+func (m *StatusBackend) onNodeStart() {
 	// tx queue manager should be started after node is started, it depends
 	// on rpc client being created
 	m.txQueueManager.Start()
-
 	if err := m.registerHandlers(); err != nil {
 		log.Error("Handler registration failed", "err", err)
 	}
-
 	if err := m.accountManager.ReSelectAccount(); err != nil {
 		log.Error("Reselect account failed", "err", err)
 	}
 	log.Info("Account reselected")
-
-	close(backendReady)
 	signal.Send(signal.Envelope{
 		Type:  signal.EventNodeReady,
 		Event: struct{}{},
 	})
 }
 
 // StopNode stop Status node. Stopped node cannot be resumed.
-func (m *StatusBackend) StopNode() (<-chan struct{}, error) {
+func (m *StatusBackend) StopNode() error {
 	m.Lock()
 	defer m.Unlock()
+	return m.stopNode()
+}
 
-	if m.nodeReady == nil {
-		return nil, node.ErrNoRunningNode
+func (m *StatusBackend) stopNode() error {
+	if !m.IsNodeRunning() {
+		return node.ErrNoRunningNode
 	}
-	<-m.nodeReady
-
 	m.txQueueManager.Stop()
 	m.jailManager.Stop()
-
-	nodeStopped, err := m.nodeManager.StopNode()
+	err := m.nodeManager.StopNode()
 	if err != nil {
-		return nil, err
+		return err
 	}
-
-	backendStopped := make(chan struct{}, 1)
-	go func() {
-		<-nodeStopped
-		m.Lock()
-		m.nodeReady = nil
-		m.Unlock()
-		close(backendStopped)
-	}()
-
-	return backendStopped, nil
+	signal.Send(signal.Envelope{
+		Type:  signal.EventNodeStopped,
+		Event: struct{}{},
+	})
+	return nil
 }
 
 // RestartNode restart running Status node, fails if node is not running
-func (m *StatusBackend) RestartNode() (<-chan struct{}, error) {
+func (m *StatusBackend) RestartNode() error {
 	m.Lock()
 	defer m.Unlock()
-
-	if m.nodeReady == nil {
-		return nil, node.ErrNoRunningNode
+	if !m.IsNodeRunning() {
+		return node.ErrNoRunningNode
 	}
-	<-m.nodeReady
-
-	nodeRestarted, err := m.nodeManager.RestartNode()
+	config, err := m.nodeManager.NodeConfig()
 	if err != nil {
-		return nil, err
+		return err
 	}
-
-	m.nodeReady = make(chan struct{}, 1)
-	go m.onNodeStart(nodeRestarted, m.nodeReady) // waits on nodeRestarted, writes to backendReady
-
-	return m.nodeReady, err
+	newcfg := *config
+	m.IsNodeRunning() {
+	
+	if err := m.stopNode(); err != nil {
```

Comment:
funky spacing. is this file `gofmt`-ed?

Reply:
strange indeed, need to check why my editor didn't fix it